tests/dogfood.rs

```diff
@@ -8,22 +8,65 @@
 // except according to those terms.
 
 #[test]
+fn dogfood_runner() {
+    dogfood();
+    dogfood_tests();
```

Comment:
Is this really the right order? Shouldn't it be the other way around or did I misunderstood your comment on this?

Reply:
It looks like I was mistaken that the order in which the tests are runs makes a difference. I've been using the following shell snippet to validate that this `dogfood_runner` function fixes the indeterminism (regardless of the actual order the implementation functions are called):

```
for i in $(seq 1 10 ) ;do
echo "run $i"
cargo test --test dogfood
done
```

Whenever I use the the `dogfood_runner` function the `dogfood` case consistently fails when it should, but if I move the `#[test]` decoration from `dogfood_runner` to each of `dogfood` and `dogfood_tests` the `dogfood` case only fails when it should something roughly every 3 out of 10 runs, sometimes it just always passes. It's kind of baffling.

Reply:
Okay, coming back to this after reading through the `cargo` docs I think the indeterminism stems not from an ordering issue, but from a parallelism issue:

> When you run multiple tests, by default they run in parallel using threads. This means the tests will finish running faster so you can get feedback quicker on whether or not your code is working. Because the tests are running at the same time, make sure your tests don’t depend on each other or on any shared state, including a shared environment, such as the current working directory or environment variables.

Both test cases do set the current working directory on the test process itself, but should instead be setting the current working directory only on the child process they are executing. I'm about to push a change to fix this.